src/d3a/models/myco_matcher/myco_external_matcher.py

```diff
@@ -107,12 +110,13 @@ def match_recommendations(self, message: Optional[Dict] = None) -> None:
         if response_data["status"] != "success":
             self.myco_ext_conn.publish_json(channel, response_data)
             return
-        for recommendation in list(
-                filter(lambda record: record["status"] == "success",
-                       response_data["recommendations"])):
+        for recommendation in response_data["recommendations"]:
             try:
-                market = self.markets_mapping.get(recommendation["market_id"])
+                if recommendation["status"] != "success":
+                    continue
                 recommendation.pop("status")
+                market = self.area_markets_mapping.get(
+                    f"{recommendation['market_id']}-{recommendation['time_slot']}")
```

Comment:
shouldn't this be
```suggestion
                market = self.area_markets_mapping.get(
                    f"{recommendation['area_uuid']}-{recommendation['time_slot']}")
```
as the recommendation now contains the area_uuid?